src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1845,29 +1845,81 @@ private FlagsExpression<FileAttributes> FormatAttributeSwitchParameters()
-            if (fileInfo == null)
-                return string.Empty;
+                var mode = new char[]
+                {
+                    isReparsePoint || isHardLink ? 'l' : isDirectory ? 'd' : '-',
+                    fileAttributes.HasFlag(FileAttributes.Archive)   ? 'a' : '-',
+                    fileAttributes.HasFlag(FileAttributes.ReadOnly)  ? 'r' : '-',
+                    fileAttributes.HasFlag(FileAttributes.Hidden)    ? 'h' : '-',
+                    fileAttributes.HasFlag(FileAttributes.System)    ? 's' : '-',
+                };
+                return new string(mode);
+            }
+
+            return instance?.BaseObject is FileSystemInfo fileInfo
+                ? ToModeString(fileInfo)
+                : string.Empty;
+        }
+
+        /// <summary>
+        /// Provides a NameString property for FileSystemInfo.
+        /// </summary>
+        /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>Name if a file or directory, Name -> Target if symlink.</returns>
+        public static string NameString(PSObject instance)
+        {
+            return instance?.BaseObject is FileSystemInfo fileInfo
+                ? InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileInfo)
+                    ? $"{fileInfo.Name} -> {InternalSymbolicLinkLinkCodeMethods.GetTarget(instance).First()}"
+                    : fileInfo.Name
+                : string.Empty;
+        }
 
-            char[] mode = new char[6];
-            mode[0] = (fileInfo.Attributes & FileAttributes.Directory) == FileAttributes.Directory ? 'd' : '-';
-            mode[1] = (fileInfo.Attributes & FileAttributes.Archive) == FileAttributes.Archive ? 'a' : '-';
-            mode[2] = (fileInfo.Attributes & FileAttributes.ReadOnly) == FileAttributes.ReadOnly ? 'r' : '-';
-            mode[3] = (fileInfo.Attributes & FileAttributes.Hidden) == FileAttributes.Hidden ? 'h' : '-';
-            mode[4] = (fileInfo.Attributes & FileAttributes.System) == FileAttributes.System ? 's' : '-';
-            // Mark the last bit as a "l" if it's a reparsepoint (symbolic link or junction)
-            // Porting note: these need to be handled specially
-            bool isReparsePoint = InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileInfo);
-            bool isHardLink = InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileInfo);
-            mode[5] = isReparsePoint || isHardLink ? 'l' : '-';
+        /// <summary>
+        /// Provides a LengthString property for FileSystemInfo.
+        /// </summary>
+        /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>Length as a string.</returns>
+        public static string LengthString(PSObject instance)
+        {
+            return instance?.BaseObject is FileInfo fileInfo
+                ? fileInfo.Attributes.HasFlag(FileAttributes.Offline)
+                    ? $"({fileInfo.Length})"
```

Comment:
Please use explicit formatting with explicit Culture.
We have an Issue to review using Culture in all code base and this help us.

Reply:
It's just an integer. When would culture matter with an integer?

Reply:
Besides, this is for UI formatting, so we want currentculture anyway, don't we?

Reply:
There may be exceptions.
In any case, it would be more convenient for us to have one pattern - we are thinking about conducting an audit of using Cultures and this would facilitate our work.